### PR TITLE
feat: Add EventViewModel extracting related code from MainPresenter

### LINKED ISSUE #1122: Create EventViewModel and extract related code from MainPresenter
### Feature Request

**Feature Description**

Create EventViewModel and extract related code from MainPresenter

**Would you like to work on the issue?**

Yes

**Parent issue**

#894

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #1122

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: 
- Add EventViewModel extracting related code from MainPresenter
- Adds an ActionLiveData in order to send one time action from ViewModel to View.(Reference: Google MVVM samples)
- As a consequence, the dashboard no longer overrides the fragments.

Screenshots for the change: 
videotogif_2018 06 22_21 10 43

**Note**
Tests will be added in a separate PR as this PR was getting huge. One more PR is needed in order to completely remove the MainPresenter. Tests will be added along with that PR.

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/common/livedata/ActionLiveData.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import android.support.annotation.MainThread;
+import android.support.annotation.Nullable;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * A wrapper over MutableLiveData in order to send a one time action from ViewModel.
+ * @param <T> Data type of the LiveData.
+ */
+public class ActionLiveData<T> extends MutableLiveData<T> {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/livedata/ActionLiveData.java]: Is it different from SingleEventLiveData

**Comment by srv-twry** [comment]
@mishuvs @sridharjajoo @Masquerade0097 Please pull it locally and test it as it is an important refactor. I have tested it already but would appreciate others testing it. Make sure to rotate the screen and verify that the behaviour is correct i.e. the dashboard doesn't appear when you're rotating the screen in Tracks fragment.

PS: 
Updating the tests to fix the build.

File: app/src/main/java/org/fossasia/openevent/app/common/livedata/ActionLiveData.java (hunk 1 above)
Comment (srv-twry) [reply app/src/main/java/org/fossasia/openevent/app/common/livedata/ActionLiveData.java]: No, do we have a SingleEventLiveData already in the project ?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/common/livedata/ActionLiveData.java]: No, it should be renamed to that and also Hungarian notation should be changed

Comment (srv-twry) [reply app/src/main/java/org/fossasia/openevent/app/common/livedata/ActionLiveData.java]: Oh, alright.

**Pull request merged**